### PR TITLE
Update substrate somewhat

### PR DESCRIPTION
**Pull request by CodeSandwich**
Updates substrate to the commit right before switch of weight from u32 to u64. This is the most we can do without introducing a breaking change. Part of https://github.com/radicle-dev/radicle-registry/issues/463

### CODE REVIEW

**Comment by NunoAlexandre** [comment]
Lookgs good to me, though I'd prefer us to update the whole way. I understand it means starting over, but we will have to do it anyway at some point. The earlier the better, it seems to me. What do you recon?

**Comment by CodeSandwich** [comment]
I think that we don't need to update in one huge leap. There are a lot of changes already, it will be easier to manage them if we don't postpone them indefinitely.

**Comment by NunoAlexandre** [comment]
> I think that we don't need to update in one huge leap. There are a lot of changes already, it will be easier to manage them if we don't postpone them indefinitely.

That's assuming that by including the breaking change that we will take much longer to update. You might be right.

**Comment by CodeSandwich** [comment]
@geigerzaehler The CLI had to be reorganized for the new substrate API. Could you take a look as well at least to be aware of the changes?

**Comment by NunoAlexandre** [comment]
> @geigerzaehler The CLI had to be reorganized for the new substrate API. Could you take a look as well at least to be aware of the changes?

what do you mean by organized?

**Comment by CodeSandwich** [comment]
> what do you mean by reorganized?

The substrate CLI API has been changed, most importantly the (SubstrateCli)[https://substrate.dev/rustdocs/v2.0.0-alpha.8/sc_cli/trait.SubstrateCli.html] trait has been added, which forced some code to be pushed to different places.

File: node/src/cli.rs [hunk 1]
```diff
@@ -96,54 +95,74 @@ pub struct Arguments {
...
+    fn support_url() -> &'static str {
+        "http://github.com/radicle-dev/radicle-registry/issues"
+    }
+
+    fn copyright_start_year() -> i32 {
+        2019
+    }
+
+    fn executable_name() -> &'static str {
+        "radicle-registry"
```
Comment (geigerzaehler) [inline node/src/cli.rs:124]:
```suggestion
        "radicle-registry-node"
```
Ideally we would get this from cargo with `env!` but I couldn’t find the right environment variable.

Comment (CodeSandwich) [reply node/src/cli.rs]: Done: https://github.com/radicle-dev/radicle-registry/pull/465/commits/8d76d044507e56606313587ba9af5566bd0a48d2 :smiley:

File: node/src/cli.rs [hunk 2]
```diff
@@ -105,11 +105,11 @@ impl SubstrateCli for Cli {
     }
 
     fn description() -> &'static str {
-        "Radicle Registry Node"
+        env!("CARGO_PKG_DESCRIPTION")
```
Comment (NunoAlexandre) [inline node/src/cli.rs:108]: can this collide with other rust projects on the same machine?

Comment (CodeSandwich) [reply node/src/cli.rs:108]: I don't think so, this is only a part of a help message. This variable isn't magical either, it's set to a `Cargo.toml`'s `description`.

**Comment by CodeSandwich** [comment]
FYI, may come in handy in the future :wink:: https://doc.rust-lang.org/cargo/reference/environment-variables.html

**Pull request merged**